### DISCUSSION TITLE
Conditional end gcode

### DISCUSSION THREAD

**Discussion by Tinchus2009** [supermerill/SuperSlicer#1680] [Q&A]
Hello. A newbie here in terms of programming.

Im trying to create a conditional gcode so I cand cool down my enclosed heated chamber slowly once the print has finished.
So I was thinking if it is posible to create a loop in order to do this and avoid to have to make it manually.

For example if my heated chamber is at 100 degrees, once print is finished I want to cool it down slowly, 2 degrees each step, each step waiting 4 minutes in each step before reducing the temp 2 more degrees. I tried but failed doing something like this:

while {(chamber_temperature[0] >50)}
          M141 P0 S{(chamber_temperature[0] >50)} ? M141 P0 S{chamber_temperature[0] -2} M141 P1 S{chamber_temperature[0] -2}  M116 G4 S240 : M141 P0 S0 M141 P1 S0 

This is not working. What this code gives me is this gocode:

while true
          M141 P0 Strue ? M141 P0 S98 M141 P1 S98  M116 G4 S240 : M141 P0 S0 M141 P1 S0 

Anyone can help on creating this code?
Thanks in advance

**Comment by supermerill** [comment]
while isn't possible as you can't store any value.

you have to do something like:
```
{if chamber_temperature[0] >2}M141 P0 S{chamber_temperature[0] - 2}
M116 G4 S240{endif}
{if chamber_temperature[0] >4}M141 P0 S{chamber_temperature[0] - 4}
M116 G4 S240{endif}
{if chamber_temperature[0] >6}M141 P0 S{chamber_temperature[0] - 6}
M116 G4 S240{endif}
{if chamber_temperature[0] >8}M141 P0 S{chamber_temperature[0] - 8}
M116 G4 S240{endif}
{if chamber_temperature[0] >10}M141 P0 S{chamber_temperature[0] - 10}
M116 G4 S240{endif}
{if chamber_temperature[0] >12}M141 P0 S{chamber_temperature[0] - 12}
M116 G4 S240{endif}
.
.
.
{if chamber_temperature[0] >48}M141 P0 S{chamber_temperature[0] - 48}
M116 G4 S240{endif}
{if chamber_temperature[0] >50}M141 P0 S{chamber_temperature[0] - 50}
M116 G4 S240{endif}
M141 P0 S0

```

**Comment by Tinchus2009** [comment]
ok. But then Im making an extra unnecessary coding, because it would be more simple to just write each gcode line with the reduced temperature setting for each step.

**Reply by supermerill** [reply]
I'm using the if, to be able to adapt to every possible chamber temp.
Thinking again, I think it's better to stop at 25° and not 0° as it won't be possible to go below ambient temp?
(if so, it's `{... > XX} .... [0] - XX + 25}` )

**Reply by Tinchus2009** [reply]
we should stop at chamber temp of 40, almosto imposible to get down that once all metal case is hot coming from 100 degrees or more (my chamber is capable of 230 degrees celsius)
I will look into other solution then, using the if aprouch from the slicer looks like not eficient. I would need to create 100 lines of if statements if I want to come down from 200 dregrees to 40 ! jajajaja

May be in the future a similar feature could be added, having the ability to use a for loop could be very useful

**Reply by supermerill** [reply]
Yes, i plan to add a more powerful scripting thing, but it's far from ready.
